code/modules/events/apc_short.dm

```diff
@@ -27,51 +27,105 @@
 	return TRUE
 
 /proc/power_failure(announce=TRUE)
-	var/list/skipped_areas_apc = list(
-		/area/engine/engineering,
-		/area/turret_protected/ai)
-
-	if(announce)
-		GLOB.event_announcement.Announce("Overload detected in [station_name()]'s powernet. Engineering, please repair shorted APCs.", "Systems Power Failure", new_sound = 'sound/AI/attention.ogg')
-
-	// break APC_BREAK_PROBABILITY% of all of the APCs on the station
-	var/affected_apc_count = 0
-	for(var/thing in GLOB.apcs)
-		var/obj/machinery/power/apc/C = thing
-		// skip any APCs that are too critical to disable
-		var/area/current_area = get_area(C)
-		if((current_area.type in skipped_areas_apc) || !is_station_level(C.z))
-			continue
-		// if we are going to break this one
-		if(prob(APC_BREAK_PROBABILITY))
-			// if it has internal wires, cut the power wires
-			if(C.wires)
-				if(!C.wires.is_cut(WIRE_MAIN_POWER1))
-					C.wires.cut(WIRE_MAIN_POWER1)
-				if(!C.wires.is_cut(WIRE_MAIN_POWER2))
-					C.wires.cut(WIRE_MAIN_POWER2)
-			// if it was operating, toggle off the breaker
-			if(C.operating)
-				C.toggle_breaker()
-			// no matter what, ensure the area knows something happened to the power
-			current_area.power_change()
-			affected_apc_count++
-	log_and_message_admins("APC Short event shorted out [affected_apc_count] APCs.")
+	switch(alert("What Would You Like to Do?", "Make All Areas Unpowered", "Depower all APCs", "Short out APCs"))
+		if("Depower all APCs")
+			var/list/skipped_areas_apc = list(
+			/area/engine/engineering,
+			/area/turret_protected/ai)
```

Comment:
Spacing.
```suggestion
			var/list/skipped_areas_apc = list(
				/area/engine/engineering,
				/area/turret_protected/ai)
```

Reply:
Can we move this declaration to above the `switch()` statement so we only declare it once?

Reply:
`/proc/power_failure(announce=TRUE)
	var/list/skipped_areas_apc = list(
		/area/engine/engineering,
		/area/turret_protected/ai)
	switch(alert("What Would You Like to Do?", "Make All Areas Unpowered", "Depower all APCs", "Short out APCs"))`

Reply:
Also, do we want to add Grav Gen to the list?

Reply:
>Can we move this declaration to above the switch() statement so we only declare it once?

Yes.
>Also, do we want to add Grav Gen to the list?

I would just leave it as is.